server/src/test/java/org/elasticsearch/index/engine/InternalEngineTests.java

```diff
@@ -5258,7 +5388,8 @@ public void testRestoreLocalHistoryFromTranslog() throws IOException {
             final EngineConfig engineConfig;
             final SeqNoStats prevSeqNoStats;
             final List<DocIdSeqNoAndSource> prevDocs;
-            try (InternalEngine engine = createEngine(store, createTempDir(), globalCheckpoint::get)) {
+            InternalEngine engine = createEngine(store, createTempDir(), globalCheckpoint::get);
```

Comment:
NIT: Perhaps use a more specific variable name to prevent the need for `super.engine` later?